cache.go

```diff
@@ -28,29 +28,39 @@ const (
 // Returned when a resource doesn't exist
 var ErrNotFoundInCache = errors.New("Not found in cache")
 
-// Cache provides a storage mechanism for cached Resources
-type Cache struct {
+type Cache interface {
+	Header(key string) (Header, error)
+	Store(res *Resource, keys ...string) error
+	Retrieve(key string) (*Resource, error)
+	Invalidate(keys ...string)
+	Freshen(res *Resource, keys ...string) error
+}
+
+// cache provides a storage mechanism for cached Resources
+type cache struct {
 	fs    vfs.VFS
 	stale map[string]time.Time
 }
 
+var _ Cache = (*cache)(nil)
```

Comment:
It's a compile time check that the type `*cache` implements the `Cache` interface.